src/Components/RelevanceInspector.tsx

```diff
@@ -47,6 +47,9 @@ export default class RelevanceInspector extends React.Component {
     this.headlessRelevanceInspector.subscribe(() => this.updateState());
   }
 
+  componentWillUnmount() {
+    this.headlessRelevanceInspector.subscribe(() => {});
+  }
```

Comment:
If you're trying to unsubscribe the relevance inspector, you'll have to save the function returned by the original subscribe call in `componentDidMount` and then call that function to unsubscribe.

Some other relevance inspector comments, because this is the first time I'm seeing the code (edit: found out in the scrum today that this didn't come from you so no worries, I've still got a lot to say though 😄 ):
- I don't understand the purpose of the second toggle to execute/not execute a query when toggling the debug. In the current code, this switch is hidden so it can't even be toggled, but even if I edit the sandbox to show it, the toggles don't work as expected. Additionally, I don't see this as a common use case/necessary at all, because you don't need to execute a new search for the information to appear in the state. I think you should just completely get rid of `hideExecuteQuery` and `executeQueryOnChange`.
- Throughout this file, you refer to `this.headlessRelevanceInspector.state` quite a bit. Then what was the point of saving that state to the variable `this.state`? You should replace all instances of the former with the latter.
- There's a lot going on in all the different functions in this file. They're not clearly named either so I had a really hard time following along and seeing intuitively which function does what. Most of the code inside of them has redundancies too or things that can be cleaned up.

I think we need to rollback this component from all of our sandboxes and redo it, because right now it's a tangled mess. I haven't even gotten to the _really_ problematic part yet, which is that we're making duplicate controllers in order to make the button available on each result. My instinct is that this is a big no-no. My suggestion for now would be to have the relevance inspector simply log its information to the console when debug is enabled, like it does in the React sample, because that should be a very straightforward component, and we can worry about the modal and the individual results later, when we have time to plan the feature and discuss best practices with the ui-kit team.